docs/snapshot-restore.md

```diff
@@ -0,0 +1,134 @@
+# Snapshot and Restore Tasks
+
+Rummager defines several rake tasks to assist in
+
+  * restoring the search indexes to an earlier state
+  * rebuilding indexes with updated popularity data
+  * mirroring the search indexes between environments
+
+This is built on top of elasticsearch's snapshot/restore functionality,
+documented at <https://www.elastic.co/guide/en/elasticsearch/reference/1.4/modules-snapshots.html>
+
+You can use these snapshots to set up a local copy of rummager with production
+data. [TBC...]
+
+## How snapshot/restore works in elasticsearch
+
+The Elasticsearch API allows us to create snapshots of one or more
+indexes.
+
+The snapshots live in a `repository`, which maps to some remote data store,
+such as a location in the file system, or in our case an AWS S3 bucket.
+
+Each snapshot is incremental, so successive snapshots don't necessarily use
+up extra storage, but there will be redundancy in the stored data, due to things
+like elasticsearch merging shards over time, or us rebuilding an index entirely.
+Because of this, we regularly delete old snapshots.
+
+### Index aliases
+
+Aliases are like symbolic links to one or more indices, and can be changed
+atomically.
+
+Rummager uses aliases to avoid downtime when restoring a snapshot or rebuilding
+an index.
+
+For example, during a restore operation, we follow the following steps:
+
+1. `mainstream` -> `mainstream-2016-02-25t17:07:13z-8efbb505-759a-4a34-ae0e-85aad065932d`
+2. Restore mainstream from a snapshot to a new index, `restored-mainstream-2016-02-25t17:11:04z-6d2a7dc4-a3b7-4572-b388-c59dc2c40be5`
+3. Wait for the restore to complete. Searches will continue to use the old
+index.
+3. Switch the alias: `mainstream` -> `restored-mainstream-2016-02-25t17:11:04z-6d2a7dc4-a3b7-4572-b388-c59dc2c40be5`
+4. Searches now use the new index.
+
+## Repository setup
+
+If you have an S3 bucket set up, you can create a snapshot repository yourself
+by running:
+
+```
+rake rummager:snapshot:create_repository[repository_name]
```

Comment:
Its arbitrary. The actual value we use will be defined in puppet (PR to follow...)

I guess this could be made more obvious though. Is there a better way to document these rake calls?